.github/workflows/pr-compile-check.yaml

```diff
@@ -0,0 +1,30 @@
+name: 'Pull requests compile checks'
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+  workflow_dispatch:
+
+jobs:
+  # Sanity check for compilation using older compiler on CentOS 7
+  pr-compile-centos-7:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
```

Comment:
is this needed, what's the default?

Reply:
6 hours for a job: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration

Reply:
My argument was if it is taking longer than this it should fail regardless, something is wrong. We should not wait 6 hours and it looks like you cannot set a default across the board.